### PR TITLE
Add missing `atomic` and `glemu` guards

### PR DESCRIPTION
**Pull request by oboukli**
Fix build with `-sSTRICT`

This PR is a WIP.  Feel free to review after it's flagged ready for review.

### CODE REVIEW

File: test/test_other.py [hunk 1]
```diff
@@ -10456,6 +10456,8 @@ def test_minimal_runtime_code_size(self, test_name, js, compare_js_output=False)
                            test_file('minimal_webgl/webgl.c'),
                            '--js-library', test_file('minimal_webgl/library_js.js'),
                            '-lwebgl.js',
+                           '--closure=1',
+                           '-sSTRICT',
```
Comment (sbc100) [inline test/test_other.py:10460]: Maybe leave this test as is, since its a sensitive code size test

File: src/library_glemu.js [hunk 2]
```diff
@@ -3910,5 +3910,7 @@ recordGLProcAddressGet(LibraryGLEmulation);
 
 addToLibrary(LibraryGLEmulation);
 
+#if ASSERTIONS
 assert(!(FULL_ES2 && LEGACY_GL_EMULATION), 'cannot emulate both ES2 and legacy GL');
 assert(!(FULL_ES3 && LEGACY_GL_EMULATION), 'cannot emulate both ES3 and legacy GL');
+#endif
```
Comment (sbc100) [inline src/library_glemu.js:3916]: Are you sure this is needed?  This looks a like a call to the build time `assert` which should always be available, even in release builds.

File: src/library_atomic.js [hunk 3]
```diff
@@ -4,7 +4,9 @@
  * SPDX-License-Identifier: MIT
  */
 
+#if ASSERTIONS
 assert(SHARED_MEMORY);
+#endif
```
Comment (sbc100) [inline src/library_atomic.js:9]: Ditto, I think this a build-time assert so should be fine.

**Review by sbc100 — COMMENTED** [review]
The build time assert is defined here: https://github.com/emscripten-core/emscripten/blob/da8fe5feac2532a24b5ed797c263a2b2dfaadea1/src/compiler.js#L12

File: test/test_other.py (hunk 1 above)
Comment (oboukli) [reply test/test_other.py]: Reverted

File: src/library_glemu.js (hunk 2 above)
Comment (oboukli) [reply src/library_glemu.js]: Reverted

File: src/library_atomic.js (hunk 3 above)
Comment (oboukli) [reply src/library_atomic.js]: Reverted

**Comment by sbc100** [comment]
Now that this PR or only changing test code should it just be closed?

**Pull request closed without merging**